### ISSUE TITLE
Provide an away to turn off padding log level with blanks

### ISSUE THREAD

**Issue by begoon** [hynek/structlog#598]
Currently, the log levels are padded with blanks to the length of the longest level. There is no way to turn this off.

I'm using the follow patching to achieve it.

```
    # patch to avoid level column padding with blank spaces
    version = metadata.version('structlog')
    if version == "23.2.0":
        console._longest_level = 0
    if version == "23.3.0":
        formatter = console._columns[1].formatter
        cast(structlog.dev.LogLevelColumnFormatter, formatter).width = 0
```

It would be better to an option for it.

**Comment by hynek** [comment]
As you've found yourself, I have added very extensive configuration affordances in 23.3.0. If I understand correctly, what you'd like are methods for easier live-patching of _structlog_'s configuration which is unprecedented and somewhat clashes with its design: the configuration is completely free-form so adding anything like this as a main API would run into the problem that the user might've changed the config and then what? If you build helpers like this yourself, you can make much more assumptions about the config which I can't.

You could make your own approach more robust by iterating over the columns and adding `isinstance` checks for LogLevelColumnFormatter which would also save you the `cast`.

**Comment by begoon** [comment]
Would it be simple to add a parameter to propagate the log level `width` in the constructor of `ConsoleRenderer`? Currently, it [sets it to 0](https://github.com/hynek/structlog/blob/186d2eaf7a1d78cba940bd3e99277a538894a6f2/src/structlog/dev.py#L631) without any chance to customise it. 

So when I create ConsoleRender, I would just do:
```
structlog.dev.ConsoleRenderer(
    pad_event=0,
    sort_keys=False,
    log_level_padding=False, ### <------- THIS NEW OPTION, False turns padding off.
)
```

**Comment by hynek** [comment]
Yeah, that makes sense.

But that's the wrong line to look at – 0 means no padding and it's `LogLevelColumnFormatter` that you want to change.

(also you absolutely already _can_ customize it, since you can pass your own `columns` – the lines further back are just a fallback from https://github.com/hynek/structlog/blob/186d2eaf7a1d78cba940bd3e99277a538894a6f2/src/structlog/dev.py#L586)

**Comment by begoon** [comment]
I have addressed your comments in the PR. Hope, that worked out.

**Comment by hynek** [comment]
fixed by #599
